src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Protocol/DelegatedTypes.cs

```diff
@@ -9,6 +9,10 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
 
+internal record DelegatedDiagnosticParams(
+    VersionedTextDocumentIdentifier HostDocument,
+    RazorLanguageKind ProjectedKind) : IDelegatedParams;
```

Comment:
Annoying that the `ProjectedKind` is needed. Does the pull diagnostics handler actually get much benefit out of inheriting from the `AbstractRazorDelegatingEndpoint`?

Reply:
I feel like the more egregious issue is what it makes us do to all other implementors of `AbstractRazorDelegatingEndpoint` (we have to make `Projection` nullable, and check that everywhere else). I'm down to drop `AbstractRazorDelegatingEndpoint` for this if you're on board.

Reply:
yea, if it's not helping just drop it :)